### ISSUE TITLE
Should we add 'assumption test' for Cox Hazard model?

### ISSUE THREAD

**Issue by chandryou** [OHDSI/CohortMethod#84]
Distriburted Research Network does not gather the whole statistical attributes, so a researcher cannot know if there was important violation of proportional hazard assumption, which may lead to wrong and misleading estimates [ref](https://www.ncbi.nlm.nih.gov/pubmed/18596734). 

Should we add 'assumption test' for Cox Hazard model, whose result a researcher can gather, too?

**Comment by schuemie** [comment]
Agreed. We should compute the Schoenfeld residuals to test proportionality.

**Comment by msuchard** [comment]
Schoenfeld $r_ij$ residuals for failure $i$ and covariate $j$ may not be terribly effective as plotting $r_ij$ as a function of covariate $j$ should be centered about 0.  However, we only have two discrete values for covariate $j$ (the treatment effect), so this may be hard to see.  The alternative is to plot as a function of failure time.  

One will need to port the code from `cox.zph()` into `CohortMethod` or `Cyclops`

**Comment by msuchard** [comment]
Another approach is to plot `log(-log(S(t)))` where `S(t)` is the survival function estimate that `Cyclops` (already? I believe) provides.

**Comment by msuchard** [comment]
BTW, the formula for the Schoenfeld residuals is super-simple for a binary covariate and a time-invariant model with Breslow-like ties (which we are using):

https://documentation.sas.com/?cdcId=pgmsascdc&cdcVersion=9.4_3.2&docsetId=statug&docsetTarget=statug_phreg_details52.htm&locale=en

**Comment by SulevR** [comment]
Is it implemented now? Not fully clear from this issue whether this is still under consideration.

**Comment by schuemie** [comment]
I'd still like to see it added, but not as a go/no go diagnostic. I don't believe the proportional hazards assumption is required for the HR to be useful.

I do think the right way to add this is to implement it in Cyclops. I don't think it is scalable to simply implement by re-fitting the model using `coxph`.
